src/annotator/features.ts

```diff
@@ -1,48 +1,38 @@
 import { TinyEmitter } from 'tiny-emitter';
 
 import { warnOnce } from '../shared/warn-once';
-
-/**
- * @typedef {import('../types/annotator').FeatureFlags} IFeatureFlags
- */
+import type { FeatureFlags as IFeatureFlags } from '../types/annotator';
 
 /**
  * List of feature flags that annotator code tests for.
- *
- * @type {string[]}
  */
 const annotatorFlags = ['html_side_by_side', 'styled_highlight_clusters'];
 
 /**
  * An observable container of feature flags.
- *
- * @implements {IFeatureFlags}
  */
-export class FeatureFlags extends TinyEmitter {
+export class FeatureFlags extends TinyEmitter implements IFeatureFlags {
+  /** Map of flag name to enabled state. */
+  private _flags: Map<string, boolean>;
+  private _knownFlags: string[];
+
   /**
-   * @param {string[]} knownFlags - Test seam. This is a list of known flags
-   *   used to catch mistakes where code checks for an obsolete feature flag.
+   * @param knownFlags - Test seam. This is a list of known flags used to catch
+   *        mistakes where code checks for an obsolete feature flag.
    */
-  constructor(knownFlags = annotatorFlags) {
+  constructor(knownFlags: string[] = annotatorFlags) {
     super();
 
-    /**
-     * Map of flag name to enabled state.
-     *
-     * @type {Map<string, boolean>}
-     */
-    this._flags = new Map();
+    this._flags = new Map<string, boolean>();
```

Comment:
This reminds me to ask! Where are we on making a decision about moving away from `_`-prefixed private attributes?

Reply:
There has been no decision as far as I know, so I have been keeping the same syntax.